org.metaborg.spoofax.meta.core/src/main/java/org/metaborg/spoofax/meta/core/pluto/build/StrIncrFrontEnd.java

```diff
@@ -0,0 +1,165 @@
+package org.metaborg.spoofax.meta.core.pluto.build;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Path;
+
+import org.apache.commons.vfs2.FileObject;
+import org.metaborg.core.MetaborgException;
+import org.metaborg.core.MetaborgRuntimeException;
+import org.metaborg.core.action.EndNamedGoal;
+import org.metaborg.core.build.BuildInput;
+import org.metaborg.core.build.BuildInputBuilder;
+import org.metaborg.core.language.ILanguageImpl;
+import org.metaborg.spoofax.core.build.ISpoofaxBuildOutput;
+import org.metaborg.spoofax.core.build.SpoofaxCommonPaths;
+import org.metaborg.spoofax.core.unit.ISpoofaxTransformUnit;
+import org.metaborg.spoofax.meta.core.pluto.SpoofaxBuilder;
+import org.metaborg.spoofax.meta.core.pluto.SpoofaxBuilderFactory;
+import org.metaborg.spoofax.meta.core.pluto.SpoofaxBuilderFactoryFactory;
+import org.metaborg.spoofax.meta.core.pluto.SpoofaxContext;
+import org.metaborg.spoofax.meta.core.pluto.SpoofaxInput;
+import org.slf4j.helpers.MessageFormatter;
+import org.spoofax.interpreter.core.Tools;
+import org.spoofax.interpreter.terms.IStrategoList;
+import org.spoofax.interpreter.terms.IStrategoTerm;
+import org.sugarj.common.FileCommands;
+
+import com.google.common.collect.Iterables;
+
+import build.pluto.BuildUnit.State;
+import build.pluto.dependency.Origin;
+import io.usethesource.capsule.BinaryRelation;
+
+public class StrIncrFrontEnd extends SpoofaxBuilder<StrIncrFrontEnd.Input, StrIncrFrontEnd.Output> {
+    private static final String COMPILE_GOAL_NAME = "Compile";
+    private static final String STRATEGO_LANG_NAME = "StrategoSugar";
+
+    public static class Input extends SpoofaxInput {
+        private static final long serialVersionUID = 1548589152421064400L;
+
+        public final SpoofaxCommonPaths paths;
```

Comment:
`SpoofaxCommonPaths` should not be passed as an input, since it is not `Serializable` nor implements `equals` or `hashCode`. Use the `paths` field of `SpoofaxBuilder` instead.